### PR TITLE
[Feature](create_table) Support create table with random distribution to avoid data skew

### LINKED ISSUE #8028: [Feature] Support data distributed by rand() to escape data skew
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

In some scenarios, users cannot find a suitable hash key to avoid data skew, so we need to provide an additional data distribution method to avoid data skew

### Use case

CREATE TABLE table1
(
    siteid INT DEFAULT '10',
    citycode SMALLINT,
    username VARCHAR(32) DEFAULT '',
    pv BIGINT SUM DEFAULT '0'
)
AGGREGATE KEY(siteid, citycode, username)
DISTRIBUTED BY random BUCKETS 10
PROPERTIES("replication_num" = "1");

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by caiconghui**
# Proposed changes

Issue Number: close #8028 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (Yes)
3. Has document been added or modified: (Yes)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/tablet_sink.cpp [hunk 1]
```diff
@@ -683,7 +683,13 @@ Status OlapTableSink::init(const TDataSink& t_sink) {
     if (table_sink.__isset.send_batch_parallelism && table_sink.send_batch_parallelism > 1) {
         _send_batch_parallelism = table_sink.send_batch_parallelism;
     }
-
+    if (table_sink.partition.distributed_columns.empty()) {
+        if (table_sink.__isset.single_tablet_load_per_sink && table_sink.single_tablet_load_per_sink) {
```
Comment (yiguolei) [inline be/src/exec/tablet_sink.cpp:687]: Please add a comment here, the meanings of these two params.

File: be/src/exec/tablet_sink.h [hunk 2]
```diff
@@ -460,6 +462,11 @@ class OlapTableSink : public DataSink {
 
     // TODO(cmy): this should be removed after we switch to rpc attachment by default.
     bool _transfer_data_by_brpc_attachment = false;
+
+    enum FindTabletMode {
+        FIND_TABLET_EVERY_ROW, FIND_TABLET_EVERY_BATCH, FIND_TABLET_EVERY_SINK
```
Comment (yiguolei) [inline be/src/exec/tablet_sink.h:467]: Add comment and explain here

File: be/src/vec/sink/vtablet_sink.cpp [hunk 3]
```diff
@@ -117,7 +120,16 @@ Status VOlapTableSink::send(RuntimeState* state, vectorized::Block* input_block)
             continue;
         }
         _partition_ids.emplace(partition->id);
-        uint32_t tablet_index = dist_hash % partition->num_buckets;
+        if (findTabletMode != FindTabletMode::FIND_TABLET_EVERY_ROW) {
```
Comment (yiguolei) [inline be/src/vec/sink/vtablet_sink.cpp:123]: How to deal with find tablet every sink?

Comment (caiconghui) [reply be/src/vec/sink/vtablet_sink.cpp:123]: if we find first tablet for one partition, we don't clear the partition_to_tablet_map before processing every row batch

**Comment by yiguolei** [comment]
LGTM

**Comment by morningman** [comment]
I have several questions and suggestions as follows:

1. this feature is directly user-oriented, we need to have more detailed documentation to explain how to use this feature, applicable scenarios, etc. Otherwise, it is difficult for other users to understand the meaning of `Random` and `single_tablet_load_per_sink`.

2. It is recommended NOT to use those `RANDOM classes` directly which has been deprecated before. We do not strictly check whether old versions of random tables still exist in the current version. Using it directly will cause metadata problems.

3. `single_tablet_load_per_sink` is too close to the specific implementation, suggest: `load_to_single_tablet`

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**